packages/pwa-kit-dev/src/configs/webpack/config.js

```diff
@@ -221,15 +225,24 @@ const ruleForBabelLoader = (babelPlugins) => {
     }
 }
 
+const findAndReplace = (array = [], findFn = () => {}, replacement) => {
+    const clone = array.slice(0)
+    const index = clone.findIndex(findFn)
+    if (index === -1) {
+        return array
+    }
+
+    clone.splice(index, 1, replacement)
+    return clone
+}
+
 const enableReactRefresh = (config) => {
     if (mode !== development || disableHMR) {
         return config
     }
 
-    const rules = [
-        ruleForBabelLoader([require.resolve('react-refresh/babel')]),
-        ...config.module.rules.slice(1)
-    ]
+    const newRule = ruleForBabelLoader([require.resolve('react-refresh/babel')])
+    const rules = findAndReplace(config.module.rules, (rule) => rule.id === 'javascript', newRule)
```

Comment:
We discussed that in the future, we might expose functionality for end users to customize specific rules. Maybe via the `Builder` class. However, that's out side of scope for this PR.